Fix clang-tidy error introduced in PR #875

See https://github.com/solvcon/modmesh/actions/runs/27211548271/job/80341703902 and https://github.com/solvcon/modmesh/actions/runs/27211548271/job/80341703885 :

```
  /home/runner/work/modmesh/modmesh/cpp/modmesh/buffer/SimpleArray.hpp:2097:9: error: use auto when initializing with a cast to avoid duplicating the type name [modernize-use-auto,-warnings-as-errors]
   2097 |         ssize_t const dim_length = static_cast<ssize_t>(shape(0));
        |         ^~~~~~~
        |         auto
  /home/runner/work/modmesh/modmesh/cpp/modmesh/buffer/SimpleArray.hpp:2098:9: error: use auto when initializing with a cast to avoid duplicating the type name [modernize-use-auto,-warnings-as-errors]
   2098 |         ssize_t const ghost_offset = static_cast<ssize_t>(m_nghost);
        |         ^~~~~~~
        |         auto
  /home/runner/work/modmesh/modmesh/cpp/modmesh/buffer/SimpleArray.hpp:2163:13: error: use auto when initializing with a cast to avoid duplicating the type name [modernize-use-auto,-warnings-as-errors]
   2163 |             ssize_t const dim_length = static_cast<ssize_t>(m_shape[dim]);
        |             ^~~~~~~
        |             auto
  14733 warnings generated.
```

This is a real defect slipping through PR CI runs.  @ThreeMonth03 please help fix them.

Related to issue #856.

--- a/cpp/modmesh/buffer/SimpleArray.hpp
+++ b/cpp/modmesh/buffer/SimpleArray.hpp
@@ -2094,8 +2094,8 @@ private:
                             "cannot use scalar index for {}-dimensional array",
                             ndim()));
         }
-        ssize_t const dim_length = static_cast<ssize_t>(shape(0));
-        ssize_t const ghost_offset = static_cast<ssize_t>(m_nghost);
+        auto const dim_length = static_cast<ssize_t>(shape(0));
+        auto const ghost_offset = static_cast<ssize_t>(m_nghost);
         ssize_t const shifted_index = it + ghost_offset;
 
         // This overload handles scalar indexing, not slicing. Python sequence
@@ -2160,7 +2160,7 @@ private:
         shape_type normalized(idx.size());
         for (size_t dim = 0; dim < m_shape.size(); ++dim)
         {
-            ssize_t const dim_length = static_cast<ssize_t>(m_shape[dim]);
+            auto const dim_length = static_cast<ssize_t>(m_shape[dim]);
             ssize_t ghost_offset = 0;
             if (dim == 0)
             {

